homeassistant/components/livisi/coordinator.py

```diff
@@ -0,0 +1,135 @@
+        super().__init__(
+            hass,
+            LOGGER,
+            name="Livisi devices",
+            update_interval=timedelta(seconds=DEVICE_POLLING_DELAY),
+        )
+        self.config_entry = config_entry
+        self.hass = hass
+        self.aiolivisi = aiolivisi
+        self.websocket = Websocket(aiolivisi)
+        self.devices: set[str] = set()
+        self.rooms: dict[str, Any] = {}
+        self.serial_number: str = ""
+        self.controller_type: str = ""
+        self.os_version: str = ""
+        self.is_avatar: bool = False
+        self.port: str = ""
+
+    async def _async_update_data(self) -> list[dict[str, Any]]:
+        """Get device configuration from LIVISI."""
+        try:
+            return await self.async_get_devices()
+        except ClientConnectorError as exc:
+            raise UpdateFailed("Failed to get LIVISI the devices") from exc
+
+    async def async_setup(self) -> None:
+        """Set up the Livisi Smart Home Controller."""
+        if not self.aiolivisi.livisi_connection_data:
+            livisi_connection_data = {
+                "ip_address": self.config_entry.data["host"],
+                "password": self.config_entry.data["password"],
+            }
+
+            await self.aiolivisi.async_set_token(
+                livisi_connection_data=livisi_connection_data
+            )
+        controller_data = await self.aiolivisi.async_get_controller()
+        if controller_data.get("controllerType") == "Avatar":
+            self.port = AVATAR_PORT
+            self.is_avatar = True
+        else:
+            self.port = CLASSIC_PORT
+            self.is_avatar = False
+        self.serial_number = controller_data["serialNumber"]
+        self.controller_type = controller_data["controllerType"]
+        self.os_version = controller_data["osVersion"]
+
+    async def async_get_devices(self) -> list[dict[str, Any]]:
+        """Set the discovered devices list."""
+        return await self.aiolivisi.async_get_devices()
+
+    async def async_get_pss_state(self, capability: str) -> bool | None:
+        """Set the PSS state."""
+        response: dict[str, Any] = await self.aiolivisi.async_get_pss_state(
+            capability[1:]
+        )
+        if response is None:
+            return None
+        on_state = response["onState"]
+        return on_state["value"]
+
+    async def async_set_all_rooms(self) -> None:
+        """Set the room list."""
+        response: list[dict[str, Any]] = await self.aiolivisi.async_get_all_rooms()
+
+        for available_room in response:
+            available_room_config: dict[str, Any] = available_room["config"]
+            self.rooms[available_room["id"]] = available_room_config["name"]
+
+    def on_data(self, event_data: LivisiEvent) -> None:
+        """Define a handler to fire when the data is received."""
+        if event_data.onState is not None:
+            device_id_state: dict = {
+                "id": event_data.source,
```

Comment:
If we use a string in more than one place, please create a constant for it and use the constant where needed.

Reply:
I changed this.